src/plugins/vis_type_script/public/kibana_api/kibana_api.ts

```diff
@@ -193,4 +210,42 @@ export class VisTypeScriptKibanaApi {
 
     return idxObj.id as string;
   }
+
+  /**
+   * Builds es query for the search request using current this.visSearchContext
+   * @param timeField
+   * @private
+   */
+  private buildEsQuery({ timeField }: { timeField?: string }) {
+    // TODO: add getNow() and getSearchSessionId()
+    const query = this.visSearchContext.query ?? [];
+    const filters = [...(this.visSearchContext.filters ?? [])];
+    if (this.visSearchContext.timeRange && timeField) {
+      // TODO: what is a proper way to build a filter from a time range without a data view?
+      const stubDataView = {
```

Comment:
it seems this is good enough for demo. 
Alternatively we can ask user to provide `index` as part of `options`